src/modules/70-pipeline/components/PipelineStages/PipelineStages.tsx

```diff
@@ -82,34 +77,34 @@ export function PipelineStages<T = Record<string, unknown>>({
   }, [showSelectMenu])
   const selected = stages.get(type || '')
 
-  const onUseTemplate = React.useCallback(
-    (templateSummary: TemplateSummaryResponse, isCopied = false) => {
-      closeTemplateSelector?.()
-      if (getNewStageFromType) {
-        setShowMenu(false)
-        setType(templateSummary.childType)
-        if (isCopied) {
-          setStageData(getNewStageFromTemplate?.(templateSummary, true))
+  const childTypes = React.useMemo(() => {
+    return [...stages.values()].filter(item => !item.isDisabled && !!item.isTemplateSupported).map(item => item.type)
+  }, [stages])
+
+  const onOpenTemplateSelector = React.useCallback(async () => {
+    if (getTemplate) {
+      try {
+        const { template: newTemplate, isCopied } = await getTemplate({
+          templateType: 'Stage',
+          allChildTypes: childTypes
+        })
+        if (getNewStageFromType) {
+          setShowMenu(false)
+          setType(newTemplate.childType)
+          if (isCopied) {
+            setStageData(getNewStageFromTemplate?.(newTemplate, true))
+          } else {
+            setStageData(getNewStageFromType?.(newTemplate.childType || '', true))
+            setTemplate(newTemplate)
+          }
         } else {
-          setStageData(getNewStageFromType?.(templateSummary.childType || '', true))
-          setTemplate(templateSummary)
+          onSelectStage?.(defaultTo(newTemplate.childType, ''))
         }
-      } else {
-        onSelectStage?.(defaultTo(templateSummary.childType, ''))
+      } catch (_) {
+        // Do nothing.. user cancelled template selection
       }
-    },
-    [closeTemplateSelector, onSelectStage]
-  )
-
-  const onOpenTemplateSelector = React.useCallback(() => {
-    openTemplateSelector?.({
-      templateType: 'Stage',
-      allChildTypes: [...stages.values()]
-        .filter(item => !item.isDisabled && !!item.isTemplateSupported)
-        .map(item => item.type),
-      onUseTemplate
-    })
-  }, [openTemplateSelector, stages, onUseTemplate])
+    }
+  }, [getTemplate, stages])
```

Comment:
Codacy has a fix for the issue: [React Hook React.useCallback has missing dependencies: 'childTypes', 'getNewStageFromTemplate', 'getNewStageFromType', and 'onSelectStage'. Either include them or remove the dependency array. If 'getNewStageFromType' changes too often, find the parent component that defines it and wrap that definition in useCallback.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232274)
```suggestion
  }, [childTypes, getNewStageFromTemplate, getNewStageFromType, getTemplate, onSelectStage])
```